src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -2776,58 +2469,60 @@ namespace FanCoilUnits {
                     Calc4PipeFanCoil(state, FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOutMax);
                 }
                 if (std::abs(QUnitOutMax) < std::abs(QZnReq)) {
-                    FanCoil(FanCoilNum).SpeedFanSel = 3;
-                    FanCoil(FanCoilNum).SpeedFanRatSel = 1.0;
-                    Node(InletNode).MassFlowRateMax = FanCoil(FanCoilNum).MaxAirMassFlow;
+                    fanCoil.SpeedFanSel = 3;
+                    fanCoil.SpeedFanRatSel = 1.0;
+                    state.dataLoopNodes->Node(InletNode).MassFlowRateMax = fanCoil.MaxAirMassFlow;
                 }
             } else {
-                FanCoil(FanCoilNum).SpeedFanSel = 0;
+                fanCoil.SpeedFanSel = 0;
             }
 
             // meet the coil load adjusted for fan operation
-            if (UnitOn && QCoilCoolSP < (-1.0 * SmallLoad) &&
+            if (UnitOn && QCoilCoolSP < (-1.0 * DataHVACGlobals::SmallLoad) &&
                 state.dataHeatBalFanSys->TempControlType(ControlledZoneNum) != DataHVACGlobals::ThermostatType::SingleHeating) {
                 // cooling coil action, maximum cold water flow
-                mdot = FanCoil(FanCoilNum).MaxCoolCoilFluidFlow;
-                SetComponentFlowRate(state,
-                                     mdot,
-                                     FanCoil(FanCoilNum).CoolCoilFluidInletNode,
-                                     FanCoil(FanCoilNum).CoolCoilFluidOutletNodeNum,
-                                     FanCoil(FanCoilNum).CoolCoilPlantLoc);
+                mdot = fanCoil.MaxCoolCoilFluidFlow;
+                PlantUtilities::SetComponentFlowRate(
+                    state, mdot, fanCoil.CoolCoilFluidInletNode, fanCoil.CoolCoilFluidOutletNodeNum, fanCoil.CoolCoilPlantLoc);
 
                 QZnReq = state.dataZoneEnergyDemand->ZoneSysEnergyDemand(ControlledZoneNum).RemainingOutputReqToCoolSP;
-                ControlOffset = FanCoil(FanCoilNum).ColdControlOffset;
+                ControlOffset = fanCoil.ColdControlOffset;
 
                 // get the maximum output of the fcu
                 Calc4PipeFanCoil(state, FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOutMax); // call without PLR means PLR = 1
 
                 if (QUnitOutMax < QZnReq) {
                     // more cooling than required, find reduced air and water flow rate to meet the load
                     // solve for the cold water flow rate with no limit set by flow rate lockdown
-                    auto f = [&state, FanCoilNum, FirstHVACIteration, ControlledZoneNum, &FanCoil, QZnReq](Real64 const PLR) {
-                        return CalcFanCoilPLRResidual(
-                            state, PLR, FanCoilNum, FirstHVACIteration, ControlledZoneNum, FanCoil(FanCoilNum).CoolCoilFluidInletNode, QZnReq);
+                    auto f = [&state, FanCoilNum, FirstHVACIteration, ControlledZoneNum, QZnReq](Real64 const PLR) {
```

Comment:
This change is just fine.  I guess it just revealed itself because of the change to using references.  :+1: